package/MDAnalysis/lib/formats/libmdaxdr.pyx

```diff
@@ -306,8 +306,14 @@ cdef class _XDRFile:
 
         # where was I
         current_frame = state[1]
-        self.seek(current_frame - 1)
-        self.current_frame = current_frame
+        if current_frame == self.offsets.size:
```

Comment:
Switch the decision in the `if` statement to make the more common situation come first
```python
if current_frame < self.offsets.size:
     ...
elif current_frame == self.offsets.size:
    ...
else:
    #pragma: no cov
    raise RuntimeError("Invalid frame number {} > {} -- this should not happen.".format(current_frame, self.offsets.size)
```
and add defensive code.

Reply:
Add comment under which circumstances we have to do funny things – namely when we need to serialize at the end of the trajectory.